lib/fog/aws/requests/efs/delete_file_system.rb

```diff
@@ -0,0 +1,46 @@
+module Fog
+  module AWS
+    class EFS
+      class Real
+        # Delete a file system
+        # http://docs.aws.amazon.com/efs/latest/ug/API_DeleteFileSystem.html
+        # ==== Parameters
+        # * FileSystemId <~String> - ID of the file system you want to delete.
+        # ==== Response
+        # * response<~Excon::Response>
+        #   * body - Empty
+        #   * status - 204
+        def delete_file_system(options={})
+          id = options.delete(:id)
+          request({
+            :path             => "file-systems/#{id}",
+            :method           => 'DELETE',
+            :expects          => 204,
+            'CreationToken'   => options[:creation_token],
+            'PerformanceMode' => options[:performance_mode] || 'generalPurpose'
+          })
+        end
+      end
+
+      class Mock
+        def delete_file_system(options={})
+          id = options.delete(:id)
+
+          unless file_system = self.data[:file_systems][id]
+            raise Fog::AWS::EFS::NotFound.new("invalid file system ID: #{id}")
+          end
+
+          if file_system["NumberOfMountTargets"] > 0
```

Comment:
I'll add that spec, but I'm going to try to add more functionality to this PR before getting it merged.  I think I figured out how to create mount targets, so it might not need to be a mock only spec.